app/src/main/java/com/sanogueralorenzo/brexit/presentation/Presenter.kt

```
@@ -10,18 +11,14 @@ interface IPresenter<in V : IView> {
     fun detachView()
 }
 
-open class Presenter<V : IView> : IPresenter<V> {
+abstract class Presenter<V : IView> : IPresenter<V> {
 
-    var view: V? = null
-    private val compositeDisposable: CompositeDisposable = CompositeDisposable()
+    abstract val view: WeakReference<V>
```

is the WeakReference in case you forget to detach the view and you try to finish the activity holding the presenter?

---

when does the view get detached then? If you go to another screen and previous screen presenter interacts with the view will the get thing make it silently fail?

---

we detach a view because by default is a strong reference, so when we try to destroy a view if someone has a referens to it (in our case the presenter) the the garbage collector can't collect it but with a weak reference if the view gets destroyed the this weakreference will nor retain anything so the grabage collecotr will be able to remove it

---

When the Activity goes into backstack does that count as the view being destroyed?